### PR TITLE
Setup ecm-distro-tools GitHub Action

### PR DESCRIPTION
**Pull request by brooksn**
This PR introduces the GitHub action for this repository, named Setup ecm-distro-tools. It simply downloads the assets from a named release, and puts them in the PATH. Inspired by [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform).

GitHub allows any repository to publish only one action. Because users may have a variety of cases for ecm-distro-tools functionality in their workflows, this action aims to be as flexible as possible to cover future needs. 

The action will not be published to the Marketplace.

usage:
```yaml
name: Main
on: push
jobs:
  main:
    runs-on: ubuntu-latest
    steps:
    - name: setup ecm-distro-tools
      uses: rancher/ecm-distro-tools
      with:
        version: v0.23.0
      env:
        GH_TOKEN: ${{ github.token }}
    - name: release notes
      run: gen_release_notes -h
```

Providing a GitHub token is recommended to avoid rate limiting, but not required.

### CODE REVIEW

**Review by Tashima42 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -104,6 +104,29 @@ docker run --rm -it --env GITHUB_TOKEN=<TOKEN> rancher/ecm-distro-tools tag_rke2
...
+
+This repository provides the "Setup ecm-distro-tools" GitHub action. 
+It downloads the assets belonging to the specified release to a temporary directory,
+and adds the directory to the `PATH`.
+
+### Usage
+
+The action can be run on ubuntu-latest runners. 
+The `version` parameter is required.
+
```
Comment (Tashima42) [inline README.md:117]: I think you should mention that the token is recommended.

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+inputs:
+  version:
+    description: 'The tag of the ecm-distro-tools release to install.'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+  - shell: bash
+    run: |
+      TAG="${{ inputs.version }}"
```
Comment (briandowns) [inline action.yml:12]: Can you run shellcheck on this to make sure it adheres to that and our script standards?

**Comment by brooksn** [comment]
The user can specify which version of the install script to use like this:


```yaml
uses: rancher/ecm-distro-tools@v0.24.0
```

```yaml
uses: rancher/ecm-distro-tools@d3d56ddf759357299e72d779bba224f42e2852f2
```

```yaml
uses: rancher/ecm-distro-tools@v1
```

Note that the version of the install script is separate from the version of the binaries. Alternately, we could explore using omitting the `version` argument and using the same version for both with this form.

File: README.md (hunk 1 above)
Comment (brooksn) [reply README.md:118]: Added a line mentioning this. Thanks.

File: action.yml (hunk 2 above)
Comment (brooksn) [reply action.yml:12]: Ran it through shellcheck and committed a change.

**Review by Tashima42 — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**